model/Check/System/TaoLtiDeliveryKVCheck.php

```diff
@@ -0,0 +1,115 @@
+<?php
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2019 (original work) Open Assessment Technologies SA (under the project TAO-PRODUCT);
+ *
+ */
+
+namespace oat\taoSystemStatus\model\Check\System;
+
+use common_report_Report as Report;
+use oat\ltiDeliveryProvider\model\execution\implementation\KvLtiDeliveryExecutionService;
+use oat\ltiDeliveryProvider\model\execution\LtiDeliveryExecutionService;
+use oat\taoSystemStatus\model\Check\AbstractCheck;
+
+/**
+ * Class TaoLtiDeliveryKVCheck
+ * @package oat\taoSystemStatus\model\Check\System
+ * @author Aleksej Tikhanovich, <aleksej@taotesting.com>
+ */
+class TaoLtiDeliveryKVCheck extends AbstractCheck
+{
+
+    /** @var Report */
+    private $report;
+
+    /**
+     * @param array $params
+     * @return Report
+     */
+    public function __invoke($params = []): Report
+    {
+        $this->checkLtiDeliveryExecutionService();
+        return $this->prepareReport($this->report);
+    }
+
+    /**
+     * @return bool
+     */
+    public function isActive(): bool
+    {
+       return $this->checkInstalledLtiExtensions();
+    }
+
+    /**
+     * @return string
+     */
+    public function getType(): string
+    {
+        return self::TYPE_SYSTEM;
+    }
+
+    /**
+     * @return string
+     */
+    public function getCategory(): string
+    {
+        return __('System configuration');
+    }
+
+    /**
+     * @return string
+     */
+    public function getDetails(): string
+    {
+        return __('Check if Lti DeliveryExecution Service correctly configured to KV implementations');
+    }
+
+    /**
+     * @return bool
+     */
+    private function checkLtiDeliveryExecutionService() : bool
+    {
+        $ltiDeliveryExecutionService = $this->getLtiDeliveryExecutionService();
+
+        if ($ltiDeliveryExecutionService instanceof KvLtiDeliveryExecutionService) {
+            $this->report = new Report(Report::TYPE_SUCCESS, __('KeyValue storage is configured for LtiDeliveryExecutionService'));
+            return true;
+        }
+
+        $this->report = new Report(Report::TYPE_WARNING, __('KeyValue storage is not configured for LtiDeliveryExecutionService'));
```

Comment:
Please change the message to the following:
"The LTI Delivery Service ('ltiDeliveryProvider/LtiDeliveryExecution') is not configured optimally. There may be performance issues."